pkg/secrets/provider_status.go

```diff
@@ -0,0 +1,113 @@
+package secrets
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+)
+
+const (
+	statusFileMode = 0666
+	scriptFileMode = 0755
+)
+
+// statusUpdater defines an interface for recording a secret provider's
+// status, and for copying utility scripts for checking that recorded status.
+//
+// setSecretsProvided: A function that records that the secrets provider
+//                     has finished providing secrets (at least for its
+//                     initial iteration.
+// setSecretsUpdated:  A function that records that the secrets provider
+//                     has just updated the secret files or Kubernetes Secrets
+//                     with recently updated secret values retrieved from
+//                     Conjur.
+// copyScripts:        Copy utility scripts for checking provider status from
+//                     a "baked-in" container directory into a volume that is
+//                     potentially shared with application container(s).
+type statusUpdater interface {
+	setSecretsProvided() error
+	setSecretsUpdated() error
+	copyScripts() error
+}
+
+type chmodFunc func(string, os.FileMode) error
+type createFunc func(string) (*os.File, error)
+type openFunc func(string) (*os.File, error)
+
+type osFuncs struct {
+	chmod  chmodFunc
+	create createFunc
+	open   openFunc
+}
+
+var stdOSFuncs = osFuncs{
+	chmod:  os.Chmod,
+	create: os.Create,
+	open:   os.Open,
+}
+
+// fileUpdater implements the statusUpdater interface. It records provider
+// status by creating empty sentinel files.
+type fileUpdater struct {
+	providedFile  string
+	updatedFile   string
+	scripts       []string
+	scriptSrcDir  string
+	scriptDestDir string
+	os            osFuncs
+}
+
+var defaultStatusUpdater = fileUpdater{
+	providedFile:  "/conjur/status/CONJUR_SECRETS_PROVIDED",
+	updatedFile:   "/conjur/status/CONJUR_SECRETS_UPDATED",
+	scripts:       []string{"conjur-secrets-unchanged.sh"},
```

Comment:
The `conjur-secrets-provided.sh` is intended to be used in a `postStart` lifecycle hook definition by the Secrets Provider container itself. If we try to use a copy of this script in `/conjur/status/` as part of a `postStart` hook definition, Kubernetes attempts to run the (non-yet-existent) copy of the script *immediately*, before Secrets Provider has had a chance to copy the script from `/usr/local/bin/` to `/conjur/status`. So I'm skipping the copy of that particular script, just to avoid any confusion / misconfiguration.